recipes/teemo/all/conanfile.py

```diff
@@ -48,12 +49,13 @@ def layout(self):
 
     def requirements(self):
         self.requires("libcurl/7.86.0")
+        self.requires("openssl/[>=1.1 <4]", transitive_headers=True)
```

Comment:
Has this always been missing then?

Reply:
@RubenRBS 
Thank you for your review.
I want to modify requiring openssl when libcurl's with_ssl is true.

```
        self.requires("libcurl/7.86.0")
        if self.dependencies["libcurl"].options.with_ssl:
            self.requires("openssl/[>=1.1 <4]", transitive_headers=True)
```
But this code is failed with following error message.
```
ERROR: teemo/2.7: Error in requirements() method, line 53
        if self.dependencies["libcurl"].options.with_ssl:
        KeyError: "'libcurl' not found in the dependency set"
```

Is there any way to solve it?

Reply:
Are the `dependencies` even available in the `requirements` method? Or perhaps a full reference is needed?

Reply:
Yeah `self.dependencies` has no chance of being valid inside the `requirements()` method, the dependencies have not been expanded at that point (It happens after the requirements() and build_requirements() methods have been called)

Reply:
I can't actually come up with a clear way to do what you want in this case @toge, let me think for a bit :)